EssentialsDiscord/src/com/earth2me/essentials/discord/DiscordListener.java

```diff
@@ -0,0 +1,28 @@
+package com.earth2me.essentials.discord;
+
+import net.dv8tion.jda.api.events.message.guild.GuildMessageReceivedEvent;
+import net.dv8tion.jda.api.hooks.ListenerAdapter;
+import org.bukkit.Bukkit;
+
+public class DiscordListener extends ListenerAdapter {
+
+    private final EssentialsDiscord plugin;
+
+    public DiscordListener(EssentialsDiscord plugin) {
+        this.plugin = plugin;
+    }
+
+    @Override
+    public void onGuildMessageReceived(GuildMessageReceivedEvent event) {
+        if (event.isWebhookMessage() || event.getAuthor().isBot()) {
```

Comment:
Is the `event.isWebhookMessage()` even needed?
As far as I know is `event.getAuthot().isBot()` also true for webhooks, so that should work too.

Reply:
Webhooks are different than bots.

Reply:
In what way are they different? They are technically bots too, so a simple check for if they are a bot is more than enough here.

![image](https://user-images.githubusercontent.com/11576465/86548069-6a8fb500-bf3b-11ea-94fe-c97e1426795d.png)

Reply:
I think I wrote this part of the code, and I wasn't sure so I erred on the side of caution (assuming that web hooks are their own distinct entity). Perhaps it can be simplified, but in any case I don't think there are any negative consequences from this sparing the possibility that a superfluous check is made.